APIMatic.Core/Utilities/CoreHelper.cs

```diff
@@ -567,31 +567,55 @@ private static void HandleCustomType(List<KeyValuePair<string, object>> processe
 
         private static void HandlePrimitiveTypes(List<KeyValuePair<string, object>> processedParameters, KeyValuePair<string, object> kvp)
         {
-            if (kvp.Value is IList)
+            if (kvp.Value is IList list)
             {
-                var list = kvp.Value as IList;
+                HandleListParameter(processedParameters, kvp, list);
+            }
+            else if (kvp.Value is IDictionary dictionary)
+            {
+                HandleDictionaryParameter(processedParameters, kvp, dictionary);
+            }
+            else
+            {
+                // Scalar type
+                processedParameters.Add(kvp);
+            }
+        }
 
-                if (list?.Count != 0)
-                {
-                    var item = list[0];
+        private static void HandleListParameter(List<KeyValuePair<string, object>> processedParameters, KeyValuePair<string, object> kvp, IList list)
+        {
+            if (list?.Count != 0)
+            {
+                var item = list[0];
 
-                    if (item.GetType().Namespace.StartsWith("System"))
-                    {
-                        // List of scalar type
-                        processedParameters.Add(kvp);
-                    }
-                    else
-                    {
-                        // List of custom type
-                        HandleCustomType(processedParameters, kvp);
-                    }
+                HandleParameter(processedParameters, kvp, item);
+            }
+        }
+
+        private static void HandleDictionaryParameter(List<KeyValuePair<string, object>> processedParameters, KeyValuePair<string, object> kvp, IDictionary dictionary)
+        {
+            if (dictionary?.Count != 0)
+            {
+                var enumerator = dictionary.GetEnumerator();
+                if (enumerator.MoveNext())
+                {
+                    var item = ((DictionaryEntry)enumerator.Current).Value;
+                    HandleParameter(processedParameters, kvp, item);
                 }
             }
-            else
+        }
+
+        private static void HandleParameter(List<KeyValuePair<string, object>> processedParameters, KeyValuePair<string, object> kvp, object entry)
+        {
+            if (entry.GetType().Namespace.StartsWith("System"))
             {
-                // Scalar type
+                // List of scalar type
                 processedParameters.Add(kvp);
             }
+            else
```

Comment:
avoid using else block, this can be done by returning at the end of its `if` block

Reply:
done